src/core/StringBuilder.hpp

```diff
@@ -96,6 +96,17 @@ class StringBuilder final
     }
 
     /**
+     * Modifies the buffer to be a substring of the current string, beginning at start and ending at end
+     */
+    void SubString(size_t start, size_t end)
```

Comment:
all `substr` implementations return `substr` and are `const`, I'd rather see it done this way here as well.